components/BottomSheet/src/MDCBottomSheetTransitionController.m

```diff
@@ -39,95 +42,88 @@ @implementation MDCBottomSheetTransitionController
+    _presenting = presenting;
+  }
   return self;
 }
 
 #pragma mark - UIViewControllerAnimatedTransitioning
 
 - (NSTimeInterval)transitionDuration:
     (nullable __unused id <UIViewControllerContextTransitioning>)transitionContext {
-  return MDCBottomSheetTransitionDuration;
+  return MDCBottomSheetMotionSpec.transitionDuration;
 }
 
 - (void)animateTransition:(id <UIViewControllerContextTransitioning>)transitionContext {
-  // If a view in the transitionContext is nil, it likely hasn't been loaded by its ViewController
-  // yet.  Ask for it directly to initiate a loadView on the ViewController.
   UIViewController *fromViewController =
       [transitionContext viewControllerForKey:UITransitionContextFromViewControllerKey];
   UIView *fromView = [transitionContext viewForKey:UITransitionContextFromViewKey];
   if (fromView == nil) {
     fromView = fromViewController.view;
   }
+  if (fromView != nil && fromView == fromViewController.view) {
+    CGRect finalFrame = [transitionContext finalFrameForViewController:fromViewController];
+    if (!CGRectIsEmpty(finalFrame)) {
+      fromView.frame = finalFrame;
+    }
+  }
 
   UIViewController *toViewController =
-  [transitionContext viewControllerForKey:UITransitionContextToViewControllerKey];
-      UIView *toView = [transitionContext viewForKey:UITransitionContextToViewKey];
+      [transitionContext viewControllerForKey:UITransitionContextToViewControllerKey];
+  UIView *toView = [transitionContext viewForKey:UITransitionContextToViewKey];
   if (toView == nil) {
     toView = toViewController.view;
   }
+  if (toView != nil && toView == toViewController.view) {
+    CGRect finalFrame = [transitionContext finalFrameForViewController:toViewController];
+    if (!CGRectIsEmpty(finalFrame)) {
+      toView.frame = finalFrame;
+    }
+
+    if (toView.superview == nil) {
+      if (_presenting) {
+        [transitionContext.containerView addSubview:toView];
+      } else {
+        [transitionContext.containerView insertSubview:toView atIndex:0];
+      }
+    }
+  }
 
-  UIViewController *toPresentingViewController = toViewController.presentingViewController;
-  BOOL presenting = (toPresentingViewController == fromViewController) ? YES : NO;
+  // We're primarily implementing UIViewControllerAnimatedTransitioning so that we can customize the
+  // transition's duration - all of the actual animation logic happens in
+  // MDCBottomSheetPresentationController (which can't customize the duration). In order to make
+  // UIKit believe that we're actually animating something, we create a temporary bogus view and
+  // animate it for the desired duration. This ensures that UIKit's default animations animate
+  // alongside our bogus animation (and therefor the corresponding duration).
 
-  UIViewController *animatingViewController = presenting ? toViewController : fromViewController;
-  UIView *animatingView = presenting ? toView : fromView;
+  UIView *bogusView = [[UIView alloc] init];
```

Comment:
bogusView seems suspicious.  It does have bogus right there in the name 🤔 
I would expect toView to be animated in the animation block below.

Reply:
I was hesitant to animate the from/to views here because all of the animations are handled by the presentation controller, thus the bogus view. Any suggestions for an alternative approach here?